tools/nebula-spark/src/main/scala/com/vesoft/nebula/tools/connector/reader/NebulaRelation.scala

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package com.vesoft.nebula.tools.connector.reader
+
+import com.vesoft.nebula.tools.connector.{DataTypeEnum, NebulaOptions, NebulaUtils}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.sources.{BaseRelation, TableScan}
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+import org.apache.spark.sql.{Row, SQLContext}
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+private case class NebulaRelation(override val sqlContext: SQLContext, nebulaOptions: NebulaOptions)
+    extends BaseRelation
+    with TableScan {
+  private val LOG = LoggerFactory.getLogger(this.getClass)
+
+  private val labelFields: Map[String, List[StructField]] = Map()
+
+  private var datasetSchema: StructType = _
```

Comment:
why don't define it in the method ?

Reply:
Because local variables must be initialized, global variables can not.